public/app/plugins/datasource/cloudwatch/utils/datalinks.ts

```diff
@@ -77,7 +77,12 @@ function createAwsConsoleLink(
     return [group.value.replace(/:\*$/, '')]; // remove `:*` from end of arn
   });
   const logGroupNames = target.logGroupNames;
-  const sources = arns ?? logGroupNames;
+  let sources = [''];
+  if (arns && arns.length > 0) {
+    sources = arns;
+  } else if (logGroupNames) {
+    sources = logGroupNames;
+  }
```

Comment:
`arns` ends up being `[]` here even when it's empty (that is the original source of this issue) which is not null/undefined, so it's not letting `sources` fall back onto logGroupNames.
(In other words, with your suggestion, the test showing the issue will start to fail.)
Maybe there would be another better way to express the if statements? Let me know.

Otherwise:
It seems `flatMap` will return an empty array instead of undefined. Maybe there would be a better way to convert the values from `target.logGroups`?

Reply:
Sounds like we need some good unit tests! 

Also if we're nervous about it at all we can put back the old code and then bring in the feature flag for our new code.

Reply:
Wonder why we use a `flatMap` above? `logGroups` is already a flat array so `map` should do just fine, no?

Reply:
Had some trouble getting the types to work with the line ` const interpolatedGroups = sources?.flatMap(getVariableValue) ?? [];`. I kept getting the possibility that it would be an array of string OR undefined, even when filtering before map.

Here's something along the lines of what I'd like. Let me know if you can help out with fixing that bit:
```
const arns = target.logGroups?.filter((group) => {
    if (group === undefined) {
      return false;
    }
    if (group.value === undefined) {
      return false;
    }
    return true;
  });
  const logGroupNames = target.logGroupNames;
  let sources = arns?.length ? arns : logGroupNames;
  const interpolatedExpression = target.expression ? replace(target.expression) : '';
  const interpolatedGroups = sources?.flatMap(getVariableValue) ?? [];
```

Reply:
> Sounds like we need some good unit tests!

Thanks for having a look.
I started adding whole new tests for the problematic cases, but then I simply included the lines for the colliding logGroups/logGroupNames in existing tests.
Let me know if you think this doesn't cover the cases quite right or if we would prefer something separate and explicit.

Reply:
Looks good. 

To address the typing issues, you can do something like this (not tested). 
```typescript
const arns = (target.logGroups ?? []).flatMap((group) => {
    if (group.value === undefined) {
      return [];
    }
    return [group.value.replace(/:\*$/, '')]; // remove `:*` from end of arn
  });
  const logGroupNames = target.logGroupNames ?? [];
  const sources = arns?.length ? arns : logGroupNames;
  const interpolatedExpression = target.expression ? replace(target.expression) : '';
  const interpolatedGroups = sources?.flatMap(getVariableValue);
```

Generally, one should avoid using reducers as they tend to be hard to read, but in this scenario it may make more sense than using a flatmap. Up to you
```typescript
  const arns = (target.logGroups ?? []).reduce<string[]>(
    (acc, currGroup) => (currGroup?.value ? [...acc, currGroup.value.replace(/:\*$/, '')] : acc),
    []
  );
```

Reply:
Sorry miswrote the filter + map above.
How about this filter + map with the fix on the typing issues?
```typescript
const arns = (target.logGroups ?? [])
    .filter((group) => {
      if (group.value === undefined) {
        return false;
      }
      return true;
    })
    .map((group) => (group.value ?? '').replace(/:\*$/, ''));
```

Reply:
Nice! 
nit improvement:
`.filter((group) => group?.value)`